yt_dlp/extractor/tencent.py

```diff
@@ -97,21 +128,29 @@ def _extract_video_native_subtitles(self, api_response, subtitles_format):
 
         return subtitles
 
-    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
-        formats, subtitles = [], {}
-        for video_format, subtitle_format, video_quality in (
-                # '': 480p, 'shd': 720p, 'fhd': 1080p
-                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
-            api_response = self._get_video_api_response(
-                url, video_id, series_id, subtitle_format, video_format, video_quality)
+    def _extract_all_video_quality(self, api_response):
+        quality = []
+        formats_response = api_response['fl']['fi']
+        for f in formats_response:
+            quality.append(f.get('name'))
+        return quality
 
-            if api_response.get('em') != 0 and api_response.get('exem') != 0:
-                if '您所在区域暂无此内容版权' in api_response.get('msg'):
-                    self.raise_geo_restricted()
-                raise ExtractorError(f'Tencent said: {api_response.get("msg")}')
+    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
+        first_api_response = self._get_video_api_response(url, video_id, series_id, 'vtt', 'hls', 'hd')
+        self._check_api_response(first_api_response)
+        api_responses = [first_api_response]
+        quality = self._extract_all_video_quality(first_api_response)
+        for q in quality:
+            if q != 'sd' and q != 'hd':
+                api_response = self._get_video_api_response(
+                    url, video_id, series_id, 'vtt', 'hls', q)
+                self._check_api_response(api_response)
+                api_responses.append(api_response)
```

Comment:
The current yt-dlp is able to get 480p direct links, now those are missing

yt-dlp 2022.10.04
```bash
[debug] Command-line config: ['https://wetv.vip/play/7csb8h9c79p8idi/a0041yw44yk', '-F', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.10.04 [4e0511f27] (debian*) API
[debug] Python 3.10.7 (CPython 64bit) - Linux-5.19.0-26-generic-x86_64-with-glibc2.36 (glibc 2.36)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-108931-g4dda3b1653-20221104 (setts), ffprobe N-108931-g4dda3b1653-20221104, phantomjs 140337668638656, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, pyxattr-0.7.2, secretstorage-3.3.3, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] Loaded 1690 extractors
[debug] [wetv:episode] Extracting URL: https://wetv.vip/play/7csb8h9c79p8idi/a0041yw44yk
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for a0041yw44yk:
ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
0  mp4 864x362    │ https │ unknown unknown
1  mp4 864x362    │ https │ unknown unknown
2  mp4 864x362    │ https │ unknown unknown
3  mp4 864x362    │ https │ unknown unknown
4  mp4 1280x536   │ m3u8  │ unknown unknown
5  mp4 1280x536   │ m3u8  │ unknown unknown
6  mp4 1280x536   │ m3u8  │ unknown unknown
7  mp4 1280x536   │ m3u8  │ unknown unknown
```

Your branch
```bash
[debug] Command-line config: ['https://wetv.vip/play/7csb8h9c79p8idi/a0041yw44yk', '--verbose', '-F']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.02 [d83b0ad80] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 62bd9c276
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-26-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg N-108931-g4dda3b1653-20221104 (setts), ffprobe N-108931-g4dda3b1653-20221104, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1756 extractors
[wetv:episode] Extracting URL: https://wetv.vip/play/7csb8h9c79p8idi/a0041yw44yk
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading webpage
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[wetv:episode] a0041yw44yk: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for a0041yw44yk:
ID EXT RESOLUTION FPS │ PROTO │ VCODEC  ACODEC  MORE INFO
──────────────────────────────────────────────────────────
0  mp4 256x108      0 │ m3u8  │ unknown unknown ld (144P)
1  mp4 256x108      0 │ m3u8  │ unknown unknown ld (144P)
2  mp4 864x362      0 │ m3u8  │ unknown unknown hd (480P)
3  mp4 864x362      0 │ m3u8  │ unknown unknown hd (480P)
4  mp4 1280x536     0 │ m3u8  │ unknown unknown shd (720P)
5  mp4 1280x536     0 │ m3u8  │ unknown unknown shd (720P)
6  mp4 1280x536     0 │ m3u8  │ unknown unknown shd (720P)
7  mp4 1280x536     0 │ m3u8  │ unknown unknown shd (720P)
```

Reply:
This is expected because the list of qualities returned by the API will not include 4K and HDR if the API is requested without the hls format.

Reply:
Why not call both API? Removing existing formats is a regression and should only be done when there is no other choice

Reply:
@pukkandan, the extraction speed for those outside of China is pretty bad, especially if you are using proxy to unblock content. In my daily use, with proxy, it takes 15–30 seconds to extract all AVC formats, in which case, extracting AVC and HEVC would double, right?

Reply:
My opinion is to add all formats by default, and have an extract-arg to limit it. I think the current `video_codec` is too limited. We still have to download multiple m3u8s for a single codec. Something like `formats=hevc-hd,avc-fhd` would be better imo

Reply:
No, we'd better not fetch both avc and hevc, even in China, it's a burden, and too many calls to the API might get blocked.

Reply:
Alright. What do you think of the suggested change to extractor-arg?

Reply:
Do you mean change `video_codec` to `formats`? Looks good, but doesn't add much value, and I think very few people need to download videos with both encoders.

Reply:
I'm confused. Calling the API is expensive, but an option to reduce number of m3u8 downloads doesn't add value?

Reply:
Sorry, my fault, I misunderstood. It does have value, and I'll consider it when I have time.